module/zfs/zio.c

```diff
@@ -3865,14 +3886,15 @@ zio_vdev_io_assess(zio_t *zio)
 	}
 
 	/*
-	 * If a cache flush returns ENOTSUP or ENOTTY, we know that no future
-	 * attempts will ever succeed. In this case we set a persistent bit so
-	 * that we don't bother with it in the future.
+	 * If a cache flush or discard returns ENOTSUP or ENOTTY, we know that
```

Comment:
This indicates that we handle TRIM failures by setting a bit but the code doesn't seem to match the comment.

Reply:
This comment change was left over from an earliest version, I'll put it back the way it was.